### ISSUE TITLE
Detect type of post, comment, comment reply, message and stuff

### ISSUE THREAD

**Issue by minhphong306** [restfb/restfb#897]
Hi restfb team,
Is there restfb provide any way to detect webhook data is post, is comment, is comment reply or message. 
And also detect action of them (like add post, edit post, remove post, add comment,....) and type of data (like post text, post image, post image and video,...) ?
I've read the documentation but I couldn't find.
Thanks you.

**Comment by nbartels** [comment]
yes, not directly but you can check the classes used and the values in the classes. This is somehow rather a abstract explanation, so here are some examples:

_You receive a new Post._
The inner change value is a `FeedPostValue` so you see it is a Post. A comment would have a different class, we see later on. Now you like to know what's the matter with that post and you simply call the `getVerb` method. The verb may be REMOVE, ADD, EDITED, HIDE, UNHIDE or EDIT. Facebook sends sometime EDITED and sometime EDIT. 

_You receive a comment._
The inner change value is a `FeedCommentValue`. The verb is similar. 

Now you see a _photo added_ and get another class. I suggest you check the classes in the `com.restfb.types.webhook` package. And check the implementations of the `ChangeValue` interface.

Then you get all information you need. The classes are direct mappings of the JSON received from Facebook and all different types, or at least many, have different fields. So we had to make the decision to go for the classes with fields that are present in the JSON or use a wide-ranging class with all fields and may run into NPEs. We took the first way.

**Comment by minhphong306** [comment]
Thank you, Norbert. I'm using this way to detect and classify data, but it is very inconvinience way.
Can restfb team build the directly detection feature in the future ? :smile: 
I think it ís very useful.

**Comment by nbartels** [comment]
Yes, this may be useful. I'll think about it and post a proposal for the implementation here.

**Comment by minhphong306** [comment]
Thank you ^^.
